### PR TITLE
PWN-4120 Swap button status fix

### PR DESCRIPTION
**Pull request by kuznetsov**
### Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4120

### Description of Work

- Fixed swap button status by updating it even if destination is still not chosen.
- Removed duplicating code in `fillMaxAmount()`

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt [hunk 1]
```diff
@@ -176,28 +176,12 @@ class OrcaSwapPresenter(
...
-        view?.setTotalAmountTextColor(availableColor)
-        view?.showAroundValue(aroundValue)
-
-        isMaxClicked = true
-
-        val destination = destinationToken ?: return
-        calculateBestPair()
-        findValidFeePayer(CORRECT_AMOUNT, destination, desiredFeePayerToken = sourceToken)
+        val amount = sourceToken.total.formatToken()
+        setSourceAmount(amount, true)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:181]: named arg plz

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/swap/model/SwapButton.kt [hunk 2]
```diff
@@ -43,10 +43,10 @@ class SwapButton(
 
             return when {
                 !isValidAmount -> State.Disabled(R.string.main_enter_the_amount)
+                !isValidDestination -> State.Disabled(R.string.swap_choose_the_destination)
                 isPairEmpty -> State.Disabled(R.string.swap_cannot_swap_these_tokens)
                 !isEnoughBalance -> State.Disabled(R.string.swap_funds_not_enough)
                 !isEnoughToCoverExpenses -> State.Disabled(R.string.swap_insufficient_funds)
-                !isValidDestination -> State.Disabled(R.string.swap_choose_the_destination)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/swap/model/SwapButton.kt:49]: Тут надо вернуть. 
Он сначала должен на пары и баланс проверить, а потом уже на destination

File: app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt [hunk 3]
```diff
@@ -176,28 +176,12 @@ class OrcaSwapPresenter(
...
-        val availableColor = if (isMoreThanBalance) R.color.systemErrorMain else R.color.textIconSecondary
-
-        view?.setTotalAmountTextColor(availableColor)
-        view?.showAroundValue(aroundValue)
-
-        isMaxClicked = true
-
-        val destination = destinationToken ?: return
-        calculateBestPair()
-        findValidFeePayer(CORRECT_AMOUNT, destination, desiredFeePayerToken = sourceToken)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:197]: Здесь тоже разломал. 
Если пользователь нажал на max, то в случае, если солов нет, мы должен уменьшить сумму инпута

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt (hunk 3 above)
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:197]: Is this about the button status? Maybe that's another task?

Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:197]: no, it's not.
There is selection mechanism is implemented and if you fix this button, you'll break the fee payer token logic. 
So, it's not right to make it as another task, please fix it here

Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:197]: I've got you. Fixed the issue

**Review by 1sakov — APPROVED** [review]

**Pull request merged**